src/test/java/com/alvarium/annotators/vulnerability/GoModHandlerTest.java

```diff
@@ -0,0 +1,139 @@
+        goModContent.append(")");
+
+        Files.writeString(gomodFile.toPath(), goModContent.toString());
+
+        GoModHandler handler = new GoModHandler(project.getAbsolutePath());
+
+        assert handler.getFileName().equals(gomodFile.getName());
+
+        Map<String, String> packages = handler.getPackages();
+        
+        assert packages.keySet().size() == 2;
+        assert packages.get(artifactId1).equals(version1);
+        assert packages.get(artifactId2).equals(version2);
+    }
+
+    @Test
+    public void testGetPackagesWithoutRequireBlock() throws Exception {
+        File project = dir.newFolder("project");
+        File gomodFile = new File(project.toPath() + "/go.mod");
+
+        String artifactId1 = "pkg1/v1";
+        String version1 = "v1.0";
+        String artifactId2 = "pkg2/v2";
+        String version2 = "v2.0";
+
+        StringBuilder goModContent = new StringBuilder();
+        goModContent.append("module github.com/edgexfoundry/edgex-go\n");
+        goModContent.append("\n");
+        goModContent.append("go 1.20\n");
+        goModContent.append("\n");
+        goModContent.append("require " + artifactId1 + " " + version1 +"\n");
+        goModContent.append("require " + artifactId2 + " " + version2 +"\n");
+
+        Files.writeString(gomodFile.toPath(), goModContent.toString());
+
+        GoModHandler handler = new GoModHandler(project.getAbsolutePath());
+
+        assert handler.getFileName().equals(gomodFile.getName());
+
+        Map<String, String> packages = handler.getPackages();
+        
+        assert packages.keySet().size() == 2;
+        assert packages.get(artifactId1).equals(version1);
+        assert packages.get(artifactId2).equals(version2);
+    }
+
+    @Test
+    public void testGetPackagesWithMixedRequireLines() throws Exception {
+        File project = dir.newFolder("project");
+        File gomodFile = new File(project.toPath() + "/go.mod");
+
+        String artifactId1 = "pkg1/v1";
+        String version1 = "v1.0";
+        String artifactId2 = "pkg2/v2";
+        String version2 = "v2.0";
+        String artifactId3 = "pkg3/v3";
+        String version3 = "v3.0";
+        String artifactId4 = "pkg4/v4";
+        String version4 = "v4.0";
+
+        StringBuilder goModContent = new StringBuilder();
+        goModContent.append("module github.com/edgexfoundry/edgex-go\n");
+        goModContent.append("\n");
+        goModContent.append("go 1.20\n");
+        goModContent.append("\n");
+
+        goModContent.append("require " + artifactId1 + " " + version1 +"\n");
+        goModContent.append("exclude example.com/old/thing v1.2.3\n");
+        goModContent.append("require (\n");
+        goModContent.append(artifactId2 + " " + version2 +"\n");
+        goModContent.append(artifactId3 + " " + version3 +"\n");
+        goModContent.append(")\n");
+        goModContent.append("example.com/bad/thing v1.4.5 => example.com/good/thing v1.4.5\n");
```

Comment:
```suggestion
        goModContent.append("replace example.com/bad/thing v1.4.5 => example.com/good/thing v1.4.5\n");
```
Won't affect the test results, but I think a `replace` keyword is missing for a valid go.mod file.

Also, do you think we should handle these replace directives ? So for example,
```go.mod
module github.com/example/project

require (
    github.com/google/uuid v1.1.0
)

replace github.com/google/uuid v1.1.0 => github.com/my-fork/uuid v1.1.1
```

This fork will most probably not have an entry in the CVE/OSV providers, but in the case that it does do we exclude the upstream package from the vulnerability scan and use the replaced (fork) package ?

Reply:
I also added a case to handle replace statements as you mentioned to do as follows:
```java
packages.remove(oldPackage);
packages.put(newPackage, newPackageVersion);
```